File: recipes_3e/apache/mysql_status/mysql_status.php
<?php
# mysql_status.php: print some MySQL server status information

require_once "Cookbook.php";

# Helper function: execute a query, return first result row as an array

function fetch_row ($dbh, $stmt)
{
  $sth = $dbh->query ($stmt);
  return ($sth->fetch (PDO::FETCH_NUM));
}
?>

<html>
<head><title>MySQL Server Status</title></head>
<body>

<?php
# Connect to database
try
{
  $dbh = Cookbook::connect ();
}
catch (PDOException $e)
{
  die ("Cannot connect to server: "
       . htmlspecialchars ($e->getMessage ()));
}

# Retrieve status information
list ($now, $version) = fetch_row ($dbh, "SELECT NOW(), VERSION()");
# SHOW STATUS variable values are in second result column
list ($junk, $queries) =
  fetch_row ($dbh, "SHOW GLOBAL STATUS LIKE 'Questions'");
list ($junk, $uptime) =
  fetch_row ($dbh, "SHOW GLOBAL STATUS LIKE 'Uptime'");
$q_per_sec = sprintf ("%.2f", $queries/$uptime);

# Disconnect from database
$dbh = NULL;
?>

<!-- Display status report -->
<p>Current time: <?php print ($now); ?></p>
<p>Server version: <?php print ($version); ?></p>
<p>Server uptime (seconds): <?php print ($uptime); ?></p>
<!-- #@ _QUERIES_ -->
<p>Queries processed:  <?php print ($queries); ?>
 (<?php print ($q_per_sec); ?> queries/second)</p>
<!-- #@ _QUERIES_ -->

</body>
</html>

File: recipes_3e/apache/mysql_status/mysql_status.rb
#!/usr/bin/ruby
# mysql_status.rb: print some MySQL server status information

require "cgi"
require "Cookbook"

# Connect to database
dbh = Cookbook.connect

# Retrieve status information
now, version = dbh.select_one("SELECT NOW(), VERSION()")
# SHOW STATUS variable values are in second result column
queries = dbh.select_one("SHOW GLOBAL STATUS LIKE 'Questions'")[1]
uptime = dbh.select_one("SHOW GLOBAL STATUS LIKE 'Uptime'")[1]
q_per_sec = sprintf("%.2f", queries.to_f / uptime.to_f)

# Disconnect from database
dbh.disconnect

# Display status report

cgi = CGI.new("html4")

cgi.out {
  cgi.html {
    cgi.head {
      cgi.title { "MySQL Server Status" }
    } +
    cgi.body() {
      cgi.p { "Current time: " + now.to_s } +
      cgi.p { "Server version: " + version.to_s } +
      cgi.p { "Server uptime (seconds): " + uptime.to_s } +
      cgi.p { "Queries processed: " + queries.to_s + " (" +
            q_per_sec.to_s + " queries/second)" }
    }
  }
}
